Source/SwiftLintFramework/Models/Linter.swift

```diff
@@ -7,16 +7,83 @@ private struct LintResult {
     let deprecatedToValidIDPairs: [(String, String)]
 }
 
-private extension Rule {
-    static func superfluousDisableCommandViolations(regions: [Region],
-                                                    superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
-                                                    allViolations: [StyleViolation]) -> [StyleViolation] {
+private enum LintableBox {
+    case rule(Rule)
+    case remoteRule(RemoteRule)
```

Comment:
I'm not a big fan of having 2 totally different models, but it seemed a better idea than having something like `CustomRules` which gives us surprising results sometimes.